vue-app/src/plugins/Web3/connectors/MetamaskConnector.ts

```diff
@@ -1,3 +1,48 @@
+import { CHAIN_INFO } from '../constants/chains'
+
+async function setupNetwork(provider): Promise<void> {
+  const chainId = parseInt(process.env.VUE_APP_ETHEREUM_API_CHAINID || '1', 10)
+  const hexChainId = `0x${chainId.toString(16)}`
+
+  // Recommended usage of these methods
+  // https://docs.metamask.io/guide/rpc-api.html#usage-with-wallet-switchethereumchain
+  try {
+    await provider.request({
+      method: 'wallet_switchEthereumChain',
+      params: [
+        {
+          chainId: hexChainId,
+        },
+      ],
+    })
+  } catch (error) {
+    // This error code indicates that the chain has not been added to MetaMask.
+    if (error.code === 4902 && CHAIN_INFO[chainId].rpcUrl) {
+      try {
+        await provider.request({
+          method: 'wallet_addEthereumChain',
```

Comment:
Currently, this case is not testable from our local envs since it requires an https rpc url (which we don't have in localhost). And in our preview deploy we are not going to be able to test it either since we are using a default MM network (Rinkeby). In this case, it is only going to be a switch (the above method `wallet_switchEthereumChain`).

The reason I left this piece of code is that probably it comes useful when we start using Arbitrum for example, where we are gonna have a rpcUrl to be added (and is not a default one).